### PR TITLE
Add 'r' command to retry last action

### PR DESCRIPTION
**Pull request by PhilipAD**
This PR adds a 'r' command to retry the last action. Users can now easily try a different approach without exiting the program.

### CODE REVIEW

**Comment by sberney** [comment]
This adds a TODO and moves a codeblock on top of implementing the new behavior. Are those required changes?

File: scripts/main.py [hunk 1]
```diff
@@ -308,6 +292,27 @@ def parse_arguments():
...
+full_message_history = []
+result = None
+# Make a constant:
+user_input = "Determine which next command to use, and respond using the format specified above:"
+
+# Interaction Loop
+while True:
+    # Send message to AI, get response
+    command_name, arguments = interaction_loop(cfg, prompt, mem, full_message_history)
+
```
Comment (nponeccop) [inline scripts/main.py]: Don't move code blocks around. If you think it's beneficial, submit this as a separate PR. Keep your diffs small.

File: scripts/main.py [hunk 2]
```diff
@@ -345,20 +353,22 @@ def parse_arguments():
...
-    # history
-    if result is not None:
-        full_message_history.append(chat.create_chat_message("system", result))
-        print_to_console("SYSTEM: ", Fore.YELLOW, result)
-    else:
-        full_message_history.append(
-            chat.create_chat_message(
-                "system", "Unable to execute command"))
-        print_to_console("SYSTEM: ", Fore.YELLOW, "Unable to execute command")
+    if user_input != "RETRY":
```
Comment (nponeccop) [inline scripts/main.py]: This is redundant as there is a comparison above already. I suggest you to just remove the if, so that the unchanged code below is showed as unchanged in diff. Then you can submit a further improvement as a separate PR.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Reduce non-essential changes to make it easier for @Torantulino to approve and merge.

**Comment by nponeccop** [comment]
@PhilipAD Remove the unrelated changes and resolve the conflicts.

**Comment by nponeccop** [comment]
@PhilipAD There are conflicts again

**Comment by nponeccop** [comment]
@PhilipAD There are conflicts again :( We are merging constantly

**Comment by nponeccop** [comment]
Now CI failed. We finally added some checks.

**Review by nponeccop — APPROVED** [review]

**Comment by richbeales** [comment]
Please remove .DS_Store file.

File: .DS_Store
Comment (AndresCdo) [inline .DS_Store:1]: Delete the ".DS_Store" file.

**Comment by PhilipAD** [comment]
The file structure has completely changed since I originally wrote this and so closing PR.
